llvm-general/src/LLVM/General/Internal/FFI/OrcJIT.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef __LLVM_GENERAL_INTERNAL_FFI__ORC_JIT__HPP__
+#define __LLVM_GENERAL_INTERNAL_FFI__ORC_JIT__HPP__
+
+#include "LLVM/General/Internal/FFI/OrcJIT.h"
+#include "LLVM/General/Internal/FFI/Target.hpp"
+
+#include "llvm/ExecutionEngine/Orc/IRCompileLayer.h"
+#include "llvm/ExecutionEngine/Orc/JITSymbol.h"
+#include "llvm/ExecutionEngine/Orc/ObjectLinkingLayer.h"
+
+typedef llvm::orc::ObjectLinkingLayer<> *LLVMObjectLinkingLayerRef;
+typedef llvm::orc::IRCompileLayer<llvm::orc::ObjectLinkingLayer<>>
+    *LLVMIRCompileLayerRef;
+typedef llvm::orc::JITSymbol *LLVMJITSymbolRef;
+typedef llvm::orc::ObjectLinkingLayer<>::ObjSetHandleT *LLVMModuleSetHandleRef;
+
+extern "C" {
+LLVMIRCompileLayerRef
+LLVM_General_createIRCompileLayer(LLVMObjectLinkingLayerRef objectLayer,
+                                  LLVMTargetMachineRef tm);
```

Comment:
Please explain more. I haven't made such declarations before to the best of my knowledge, so it seems inconsistent to me.

What simplification do you mean?